lib/ecl/tests/ecl_sum_restart.cpp

```diff
@@ -0,0 +1,167 @@
+      test_assert_double_equal( double_vector_iget(d, j-1), (1 - i)*100 + j*1000);
+    double_vector_free(d);
+  }
+
+  ecl_sum_free(sum);
+
+  sum = ecl_sum_fread_alloc_case("CASE2", ":");
+  for (int i=0; i < params_size; i++) {
+    double_vector_type * d = ecl_sum_alloc_data_vector(sum, i+1, false);
+    test_assert_int_equal(double_vector_size(d), 4);
+    ieq(d,0,(1 - i)*100 + 1000);
+    ieq(d,1,(1 - i)*100 + 2000);
+    ieq(d,2,(1 - i)*100 + 3000);
+    ieq(d,3,(1 - i)*100 + 4000);
+    double_vector_free(d);
+  }
+
+  ecl_sum_free(sum);
+}
+
+
+void write_CASE1(bool unified) {
+  time_t start_time = util_make_date_utc( 1,1,2010 );
+  ecl_sum_type * ecl_sum = ecl_sum_alloc_writer( "CASE1" , false , unified, ":" , start_time , true , 10 , 10 , 10 );
+  double sim_seconds = 0;
+
+  int num_dates = 100;
+  double ministep_length = 86400; // seconds in a day
+
+  const ecl::smspec_node * node1 = ecl_sum_add_var( ecl_sum , "BPR" , NULL   , 1   , "BARS"    , 0.0  );
+  const ecl::smspec_node * node2 = ecl_sum_add_var( ecl_sum , "BPR" , NULL   , 2   , "BARS"    , 0.0  );
+  const ecl::smspec_node * node3 = ecl_sum_add_var( ecl_sum , "BPR" , NULL   , 3   , "BARS"    , 0.0  );
+
+  for (int report_step = 0; report_step < num_dates; report_step++) {
+      {
+        ecl_sum_tstep_type * tstep = ecl_sum_add_tstep( ecl_sum , report_step + 1 , sim_seconds );
+        ecl_sum_tstep_set_from_node( tstep , *node1 , (1 + report_step) * 100 );
+        ecl_sum_tstep_set_from_node( tstep , *node2 , (1 + report_step) * 100 + 10.0 );
+        ecl_sum_tstep_set_from_node( tstep , *node3 , (1 + report_step) * 100 + 20.0 );
+      }
+      sim_seconds += ministep_length * 3;
+  }
+  ecl_sum_fwrite( ecl_sum );
+  ecl_sum_free(ecl_sum);
+
+  verify_CASE1(num_dates);
+}
+
+
+
+
+void write_CASE2() {
+  bool unified = false;
+  write_CASE1(unified);
+  {
+    time_t start_time = util_make_date_utc( 1,1,2010 );
+    int num_dates = 4;
+    double ministep_length = 86400; // seconds in a day
+    double sim_seconds = ministep_length * 2.5 * 3;
+    ecl_sum_type * ecl_sum = ecl_sum_alloc_restart_writer( "CASE2" , "CASE1", false , true , ":" , start_time , true , 10 , 10 , 10 );
+
+    const ecl::smspec_node * node2 = ecl_sum_add_var( ecl_sum , "BPR" , NULL   , 2   , "BARS"    , 0.0  );
+    const ecl::smspec_node * node1 = ecl_sum_add_var( ecl_sum , "BPR" , NULL   , 1   , "BARS"    , 0.0  );
+
+    for (int report_step = 1; report_step <= num_dates; report_step++) {
+      {
+        ecl_sum_tstep_type * tstep = ecl_sum_add_tstep( ecl_sum , report_step + 3 , sim_seconds );
+        ecl_sum_tstep_set_from_node( tstep , *node1 , report_step*1000);
+        ecl_sum_tstep_set_from_node( tstep , *node2 , report_step*1000 + 100);
+      }
+      sim_seconds += ministep_length * 3;
+    }
+    ecl_sum_fwrite( ecl_sum );
+    ecl_sum_free(ecl_sum);
+
+    util_unlink("CASE1.SMSPEC");
+
+    verify_CASE2(num_dates);
+  }
+}
+
+int main() {
+  write_CASE2();
```

Comment:
Should we consider running all the tests here ;)

Reply:
It is recursive - case2 calls case1